### ISSUE TITLE
BlockUI requires targeted component to be unblocked during page load

### ISSUE THREAD

**Issue by JosephFranks** [primefaces/primeng#11035]
**I'm submitting a ...**  (check one with "x")
```
[x ] bug report => Search github for a similar issue or PR before submitting
[ ] feature request => Please check if request is not on the roadmap already https://github.com/primefaces/primeng/wiki/Roadmap
[ ] support request => Please do not submit support request here, instead see http://forum.primefaces.org/viewforum.php?f=35
```

**Plunkr Case (Bug Reports)**
https://stackblitz.com/edit/primeng-blockui-demo-tzxuqq

**Current behavior**
When BlockUI targets a component with [blocked] set to true during page load, the entire body of the page is blocked rather than just the targeted component.  This seems to be caused by the fact that BlockUI's this.target is undefined until ngAfterViewInit is called.

**Expected behavior**
Only the targeted component should be blocked rather than the entire body.

**Minimal reproduction of the problem with instructions**
The stackblitz I linked above (https://stackblitz.com/edit/primeng-blockui-demo-tzxuqq) is identical to the stackblitz from the docs here:
https://www.primefaces.org/primeng/showcase/#/blockui
with one exception.  I changed line 14 of app.component.ts from
`blockedPanel: boolean = false;`
to
`blockedPanel: boolean = true;`
As you can see, the entire body is blocked rather than just the #pnl target.

**Proposed Fix**
In the file primeng/src/app/components/blockui/blockui.ts, changing the following function from
```
ngAfterViewInit() {
    if (this.target && !this.target.getBlockableElement) {
        throw 'Target of BlockUI must implement BlockableUI interface';
    }
}
```
to
```
ngAfterViewInit() {
    if (this.target) {
        if (!this.target.getBlockableElement) {
            throw 'Target of BlockUI must implement BlockableUI interface';
        }

        if (this._blocked) {
            this.block();
        }
    }
}
```
seems to fix the issue.  Since this.target is no longer undefined by this point (if a target is specified), calling the block function here seems to successfully re-target the block from the body to the desired component. 

I created a PR with this code here:
#11037 

**What is the motivation / use case for changing the behavior?**
Sometimes we want to block a component until the user takes a certain action or enters certain information.  Blocking the entire body prevents them from being able to do so.  Being forced to default the targeted component to unblocked and then toggling it to blocked at the appropriate point in the component lifecycle is a hassle and can lead to expression has changed after it was checked errors if not done properly.

* **Angular version:** 
At least Angular versions 8-13

* **PrimeNG version:** 5.X
At least PrimeNG versions 8-13

* **Browser:** [Chrome | Firefox | IE | Safari | Edge]

**Comment by berkinsansal** [comment]
I developed a workaround. That workaround can be used until this issue is fixed.
You can add below code to the *.ts file of the component where you use `p-blockUI`
```
@ViewChild(BlockUI) set blockUI(blockUI: BlockUI) {
    if (blockUI) {
        blockUI.blocked = blockUI.blocked;
    }
}
```

**Comment by mertsincan** [comment]
Hi,

So sorry for the delayed response! Improvements have been made to many components recently, both in terms of performance and enhancement. Therefore, this improvement may have been developed in another issue ticket without realizing it. You can check this in the documentation. If there is no improvement on this, can you reopen the issue so we can include it in our roadmap? 
Please don't forget to add your feedback as a comment after reopening the issue. These will be taken into account by us and will contribute to the development of this feature. Thanks a lot for your understanding!

Best Regards,
